pages/instructorCourseAdminSharing/instructorCourseAdminSharing.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const router = express.Router();
+
+const path = require('path');
+const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+const { v4: uuidv4 } = require('uuid');
+const ERR = require('async-stacktrace');
+const error = require('../../prairielib/lib/error');
+const { InstructorSharing } = require('./instructorCourseAdminSharing.html');
+const sqldb = require('../../prairielib/lib/sql-db');
+const sqlLoader = require('../../prairielib/lib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+router.get('/', function (req, res, next) {
+  debug('GET /');
+
+  sqldb.queryOneRow(
+    sql.get_course_sharing_info,
+    { course_id: res.locals.course.id },
+    (err, result) => {
+      if (ERR(err, next)) return;
+      let sharing_name = result.rows[0].sharing_name;
+      let sharing_id = result.rows[0].sharing_id;
+
+      sqldb.query(sql.select_sharing_sets, { course_id: res.locals.course.id }, (err, result) => {
+        if (ERR(err, next)) return;
+
+        res.send(
+          InstructorSharing({
+            sharing_name: sharing_name,
+            sharing_id: sharing_id,
+            sharing_sets: result.rows,
+            resLocals: res.locals,
+          })
+        );
+      });
+    }
+  );
+});
+
+router.post('/', (req, res, next) => {
+  if (!res.locals.authz_data.has_course_permission_own) {
+    return next(error.make(403, 'Access denied (must be course owner)'));
+  }
+
+  if (req.body.__action === 'sharing_id_regenerate') {
+    let newSharingId = uuidv4();
+    sqldb.queryZeroOrOneRow(
+      sql.update_sharing_id,
+      { sharing_id: newSharingId, course_id: res.locals.course.id },
+      (err, _result) => {
+        if (ERR(err, next)) return;
+
+        res.redirect(req.originalUrl);
+      }
+    );
+  } else if (req.body.__action === 'sharing_set_create') {
+    // TODO: disallow duplicates!
+    sqldb.queryZeroOrOneRow(
+      sql.create_sharing_set,
+      { sharing_set_name: req.body.sharing_set_name, course_id: res.locals.course.id },
+      (err, _result) => {
+        if (ERR(err, next)) return;
+
+        res.redirect(req.originalUrl);
+      }
+    );
+  } else if (req.body.__action === 'course_sharing_set_add') {
+    sqldb.queryZeroOrOneRow(
+      sql.course_sharing_set_add,
+      { sharing_set_id: req.body.sharing_set_id, course_sharing_id: req.body.course_sharing_id },
+      (err, _result) => {
+        if (ERR(err, next)) return;
+
+        res.redirect(req.originalUrl);
+      }
+    );
+  } else if (req.body.__action === 'course_sharing_set_delete') {
+    // TODO: do we actually want to allow deleting the sharing with a course?
+    // In the future we want to do some fancy versioning stuff. How do we transition from allowing deletion
+    // to versioning questions?
```

Comment:
Here's where it gets weird: for now, since we don't have a versioning system, people could essentially just delete all the content out of the questions to un-share them, so it seems silly to not just have an un-share functionality anyway. But, later when we have some sort of versioning system, we may want people to have the ability to stop sharing future versions of question, but we can use the fact that we store versions to ensure no one's questions ever get surprisingly broken.

Reply:
You know what, I'm just going to go ahead and make it so that there is no way to un-share things for now. That's we want ultimately (when we have a good versioning system), and it will keep things simpler to deploy the MVP.